app/models/registration.rb

```diff
@@ -593,10 +590,10 @@ def last_positive_payment
       .first
   end
 
-  delegate :auto_accept_registrations?, to: :competition
+  delegate :auto_accept_preference, :auto_accept_preference_disabled?, :auto_accept_preference_bulk?, :auto_accept_preference_live?, to: :competition
 
-  def attempt_auto_accept
-    failure_reason = auto_accept_failure_reason
+  def attempt_auto_accept(expected_setting)
```

Comment:
Still not quite happy with the name. At the moment of triggering this attempt, it is the actual real mode. Nothing expected, but the actual real thing.
What you're _expecting_ is for the real mode and the competition setting to match. But the parameter here does not / should not represent that expectation. The parameter should be "what's really happening right now", and you're formulating an expectation based on this in the `failure_reason` method down below.

Reply:
The competition setting is certainly one source of truth.

The mode that the current auto accept attempt runs under is another source of truth.

The _expectation_ is that these two truths / facts match up. You're not passing this expectation to the attempt. You're passing a truth to the attempt (this auto accept run is run as bulk / run as live) and later down when you're comparing that truth to the competition setting truth, there's an expectation embedded into that comparison.

Reply:
Renamed to `required_setting` - so the calling function is asserting what setting is needed for its call to succeed